core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2515,21 +2503,35 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
           UnderFileSystem ufs = resolution.getUfs();
           if (ownerGroupChanged) {
             try {
-              ufs.setOwner(ufsUri, inode.getOwner(), inode.getGroup());
+              String owner = options.getOwner() != null ? options.getOwner() : inode.getOwner();
+              String group = options.getGroup() != null ? options.getGroup() : inode.getGroup();
+              ufs.setOwner(ufsUri, owner, group);
             } catch (IOException e) {
-              throw new AccessControlException("Could not setOwner for UFS file " + ufsUri, e);
+              throw new AccessControlException("Could not setOwner for UFS file " + ufsUri
+                  + " . Aborting the setAttribute operation in Alluxio.", e);
             }
           }
-          if (permissionChanged) {
+          if (modeChanged) {
             try {
-              ufs.setMode(ufsUri, inode.getMode());
+              ufs.setMode(ufsUri, options.getMode());
             } catch (IOException e) {
-              throw new AccessControlException("Could not setMode for UFS file " + ufsUri, e);
+              throw new AccessControlException("Could not setMode for UFS file " + ufsUri
+                  + " . Aborting the setAttribute operation in Alluxio.", e);
```

Comment:
Is it possible for the owner and group change to succeed but the mode change to fail?